File: ios/Packages/Components/Sources/Components/ProviderSignInCard.swift
import DesignSystem
import Models
import SwiftUI

public struct ProviderSignInCard: View {
    @Environment(\.appTheme) private var theme

    private let title: String
    private let description: String
    private let providers: [SwiftAuthProvider]
    private let showBackground: Bool
    private let onProviderSelected: (String) -> Void

    public init(
        title: String,
        description: String,
        providers: [SwiftAuthProvider],
        showBackground: Bool = true,
        onProviderSelected: @escaping (String) -> Void
    ) {
        self.title = title
        self.description = description
        self.providers = providers
        self.showBackground = showBackground
        self.onProviderSelected = onProviderSelected
    }

    public var body: some View {
        if showBackground {
            content
                .padding(theme.spacing.medium)
                .frame(maxWidth: .infinity, alignment: .leading)
                .background(theme.colors.surface)
                .clipShape(RoundedRectangle(cornerRadius: theme.shapes.large))
                .overlay(
                    RoundedRectangle(cornerRadius: theme.shapes.large)
                        .stroke(theme.colors.outline.opacity(0.2), lineWidth: 0.5)
                )
        } else {
            content
        }
    }

    private var content: some View {
        VStack(alignment: .leading, spacing: theme.spacing.large) {
            VStack(alignment: .leading, spacing: theme.spacing.xxSmall) {
                Text(title)
                    .textStyle(theme.typography.titleLarge)
                    .foregroundColor(theme.colors.onSurface)
                Text(description)
                    .textStyle(theme.typography.bodyMedium)
                    .foregroundColor(theme.colors.onSurfaceVariant)
            }

            VStack(spacing: theme.spacing.medium) {
                ForEach(providers) { provider in
                    ProviderButton(title: provider.label, logo: provider.logoName) {
                        onProviderSelected(provider.id)
                    }
                }
            }
        }
    }
}

#if DEBUG
    private let previewProviders: [SwiftAuthProvider] = [
        SwiftAuthProvider(id: "TRAKT", label: "Continue with Trakt", logoName: "TraktMono"),
        SwiftAuthProvider(id: "SIMKL", label: "Continue with Simkl", logoName: "SimklMono"),
    ]

    #Preview("With background") {
        ProviderSignInCard(
            title: "Connect & Sync Your Content",
            description: "Save your progress, discover new titles, and sync your content across all devices.",
            providers: previewProviders,
            onProviderSelected: { _ in }
        )
        .padding()
        .appPreview()
    }

    #Preview("No background") {
        ProviderSignInCard(
            title: "Connect & Sync Your Content",
            description: "Save your progress, discover new titles, and sync your content across all devices.",
            providers: previewProviders,
            showBackground: false,
            onProviderSelected: { _ in }
        )
        .padding()
        .appPreview()
    }
#endif

File: android-designsystem/src/main/kotlin/com/thomaskioko/tvmaniac/compose/components/ProviderSignInCard.kt
package com.thomaskioko.tvmaniac.compose.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.PreviewWrapper
import androidx.compose.ui.unit.dp
import com.thomaskioko.tvmaniac.android.designsystem.R

@Composable
public fun ProviderSignInCard(
    title: String,
    description: String,
    modifier: Modifier = Modifier,
    showBackground: Boolean = true,
    buttons: @Composable ColumnScope.() -> Unit,
) {
    if (showBackground) {
        Surface(
            modifier = modifier.fillMaxWidth(),
            color = MaterialTheme.colorScheme.surface,
            shape = RoundedCornerShape(16.dp),
            border = BorderStroke(
                width = 0.5.dp,
                color = MaterialTheme.colorScheme.outline.copy(alpha = 0.2f),
            ),
        ) {
            ProviderSignInContent(
                title = title,
                description = description,
                modifier = Modifier.padding(16.dp),
                buttons = buttons,
            )
        }
    } else {
        ProviderSignInContent(
            title = title,
            description = description,
            modifier = modifier,
            buttons = buttons,
        )
    }
}

@Composable
private fun ProviderSignInContent(
    title: String,
    description: String,
    modifier: Modifier = Modifier,
    buttons: @Composable ColumnScope.() -> Unit,
) {
    Column(
        modifier = modifier.fillMaxWidth(),
        verticalArrangement = Arrangement.spacedBy(16.dp),
    ) {
        Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
            Text(
                text = title,
                style = MaterialTheme.typography.titleLarge,
                color = MaterialTheme.colorScheme.onSurface,
                fontWeight = FontWeight.Bold,
            )
            Text(
                text = description,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        }

        Column(
            verticalArrangement = Arrangement.spacedBy(12.dp),
            content = buttons,
        )
    }
}

@ThemePreviews
@PreviewWrapper(TvManiacPreviewWrapperProvider::class)
@Composable
private fun ProviderSignInCardPreview() {
    Box(Modifier.padding(16.dp)) {
        ProviderSignInCard(
            title = "Connect & Sync Your Content",
            description = "Save your progress, discover new titles, and sync your content across all devices.",
        ) {
            ProviderButton(
                text = "Continue with Trakt",
                logo = R.drawable.ic_trakt_mono,
                onClick = {},
            )
            ProviderButton(
                text = "Continue with Simkl",
                logo = R.drawable.ic_simkl_mono,
                onClick = {},
            )
        }
    }
}
